### ISSUE TITLE
About albedo evaluation detail

### ISSUE THREAD

**Issue by Maddog241** [nv-tlabs/cosmos1-diffusion-renderer#20]
Hi, thank you for your excellent work!
I’d like to ask for some clarification regarding the evaluation of inverse rendering in your paper.
When computing the albedo PSNR, was it measured in sRGB space or in linear space? Also, was a scale factor applied beforehand to account for the scale ambiguity?
Thanks!

**Comment by Maddog241** [comment]
Thank you!

**Comment by Maddog241** [comment]
@xdobetter @zian-w 
Hello, I just evaluated the SVD model and got similar issues.
I randomly sampled 256 images in the test split of interiorverse dataset (fov85 subset), the visual quality is very good, but the evaluated PSNR is only between 13-14, while the paper reported  21.9. So I think there might be something wrong in my evaluation. 

1. I used the SVD model and ran inference on 256 images. `Inference_n_frames` is set to be 1, `Inference_n_steps=20`, and other arguments are the default args (512 x 512 resolution, etc). 
2. The images in interiorverse are in .exr format, so I firstly converted them into srgb space, ran the evaluation and saved the basecolor (also in srgb space)
3. I used metric code from https://github.com/graphdeco-inria/gaussian-splatting/blob/main/metrics.py to compute the metrics. Before sending the images into `psnr, ssim, lpips` functions, I converted them from srgb space to linear space and computed a 3-channel scaling factor.
4. For srgb-linear convertion and 3-channel scaler computing, I used code below.

```python
def compute_scale(pred, gt, eps=1e-3):
    # [B, C, H, W]
    B, C, H, W = pred.shape
    pred = pred.reshape(B, C, -1)
    gt = gt.reshape(B, C, -1)
    denom = torch.sum(pred * pred, dim=-1)
    nume = torch.sum(pred * gt, dim=-1)
    s = nume / (denom + eps)
    return s

def linear_to_srgb_tensor(linear: torch.Tensor) -> torch.Tensor:
    """
    Linear RGB -> sRGB
    linear: tensor in [0,1], shape (..., C, H, W) 或 (..., H, W, C)
    """
    threshold = 0.0031308
    low = linear * 12.92
    high = 1.055 * (torch.clamp(linear, min=0.0) ** (1/2.4)) - 0.055
    return torch.where(linear <= threshold, low, high).clamp(0, 1)

def srgb_to_linear_tensor(srgb: torch.Tensor) -> torch.Tensor:
    """
    sRGB -> Linear RGB
    srgb: tensor in [0,1], shape (..., C, H, W) 或 (..., H, W, C)
    """
    threshold = 0.04045
    low = srgb / 12.92
    high = ((srgb + 0.055) / 1.055) ** 2.4
    return torch.where(srgb <= threshold, low, high)
```

And I also performed evaluation in srgb space, but that did not make big a difference. So I guess this may not be the primary cause of such a huge performance gap. Can you please help me figure out what's wrong with my evalution? Cause I want to compare DiffusionRenderer in my paper. 

Another thing I noted is images in InteriorVerse are noisy due to ray tracing. DiffusionRenderer tends follow the noise and generate noisy outputs like the pair I paste below. Is that the probable cause? Did you explicitly handled this? 

Or can you please share the evaluation script?

I would really appreciate your help on this. Thank you very much! 

<img width="704" height="512" alt="Image" src="https://github.com/user-attachments/assets/11997894-4867-4b6e-8033-6c85873a6e6d" />

<img width="704" height="512" alt="Image" src="https://github.com/user-attachments/assets/a8dbf350-2b35-41c4-b3e3-195f1d6e8165" />

**Comment by Maddog241** [comment]
> It reminds me we pre-processed full InteriorVerse RGB with optix denoiser and tone-mapping at the start of the project, as we observed rendering artifacts from the default dataset. I’m happy to share our evaluation results if it helps a fair comparison, feel free to reach out through email!

Thank you so much! I will give it a try.
